wallet/src/repository/wallet/mod.rs

```diff
@@ -36,6 +43,75 @@ struct AccountMutation {
     utxo_removals: Vec<model::OutPtr>,
 }
 
+/// Struct that keep the unspent outputs pool and the own unspent outputs pool
+#[derive(Debug)]
+pub struct WalletUtxos<'a> {
+    pub state: &'a mut State,
+}
+
+impl<'a> UtxoFantasy for WalletUtxos<'a> {
+    fn sort_iter(&self, strategy: UtxoSelectionStrategy) -> Vec<OutputPointer> {
+        match strategy {
+            UtxoSelectionStrategy::BigFirst => sort_utxo_set(&self.state.utxo_set, true),
+            UtxoSelectionStrategy::SmallFirst => sort_utxo_set(&self.state.utxo_set, false),
+            UtxoSelectionStrategy::Random => self
+                .state
+                .utxo_set
+                .iter()
+                .map(|(o, _)| o.clone().output_pointer())
+                .collect(),
+        }
+    }
+
+    fn get_time_lock(&self, outptr: &OutputPointer) -> Option<u64> {
+        let time_lock = self
+            .state
+            .utxo_set
+            .get(&outptr.into())
+            .map(|vto| vto.time_lock);
+        let time_lock_by_used = self.state.used_utxo_set.get(&outptr.into()).copied();
+
+        match (time_lock, time_lock_by_used) {
+            (Some(a), Some(b)) => Some(std::cmp::max(a, b)),
+            (Some(a), None) => Some(a),
+            (None, Some(b)) => Some(b),
+            _ => None,
+        }
+    }
+
+    fn get_value(&self, outptr: &OutputPointer) -> Option<u64> {
+        self.state
+            .utxo_set
+            .get(&outptr.into())
+            .map(|vto| vto.amount)
+    }
+
+    fn get_included_block_number(&self, _outptr: &OutputPointer) -> Option<u32> {
+        None
```

Comment:
What are the implications if this returning `None`?

Reply:
I see this being used as

```rust
if utxos.get_included_block_number(op).unwrap() > block_number_limit {
```

So we probably want to use `unwrap_or(0)` there?

Reply:
it is only used for collaterals in the part of the node, to check the collateral age. In case of wallet it will be never used so thats the reason of the None. I could add the unwrap_or(0) for safuness but I think it would not change the current behaviour

Reply:
Ok, so if `get_included_block_number` returns `None` it means that we don't know the collateral age. In that case instead of `unwrap_or(0)` let's skip that UTXO, using something like `if x.is_none() { continue; }`